packages/client/public/root-cookie-accessor-template.html

```diff
@@ -1,141 +1,177 @@
-                            } catch(err) {
-                                if (storageAccessPermission.state === 'prompt')
-                                    parent.postMessage(JSON.stringify({ hasStorageAccess: false, cookieSet }), e.origin)
-                                else if (storageAccessPermission.state === "denied")
-                                    parent.postMessage(JSON.stringify({ hasStorageAccess: false, cookieSet, storageAccessPermission: 'denied' }), e.origin)
-                            }
-                        }
-                        break
-                    }
+      let cookiesMapObj = getCookies()
+
+      const processMessage = async (e) => {
+        origin = e.origin
+        const payload = e?.data?.payload
+        if (e?.data?.method == null) return
+        const parent = window.parent
+        const message = { id: e.data.id, success: true, data: {} }
+        switch (e?.data?.method) {
+          case 'set':
+            if (payload?.data) {
+              document.cookie = `${payload.key}=${JSON.stringify(payload.data)}; SameSite=None; Secure`
+              message.data.cookieWasSet = payload.key
+              parent.postMessage(message, e.origin)
             }
-        }
-        window.onmessage = async function(e) {
-            let allowedDomains
-            try {
-                await document.requestStorageAccess()
-                const allowedDomainCookie = document.cookie
-                    .split("; ")
-                    .find((row) => row.startsWith(`allowedDomains=`))
-                allowedDomains = allowedDomainCookie ? JSON.parse(allowedDomainCookie.split("=")[1]) : []
-            } catch (err) {
-                allowedDomains = []
+            break
+          case 'get':
+            const result = cookiesMapObj[payload.key]
+            if (result === undefined || result === '') {
+              message.success = false
+              parent.postMessage(message, e.origin)
+              break
             }
-            if (allowedDomains?.indexOf(e.origin) > -1) await processMessage(e)
-            else {
-                const response = await fetch(`<API_URL>/allowed-domains?domainToCheck=${e.origin}`)
-                if (response.status === 200) {
-                    const body = await response.json()
-                    if (body) {
-                        allowedDomains.push(e.origin)
-                        document.cookie = `allowedDomains=${JSON.stringify(allowedDomains)}; SameSite=None; Secure`
-                        await processMessage(e)
-                    }
-                } else {
-                    parent.postMessage(JSON.stringify({invalidDomain: true}), e.origin)
+            message.data = result
+            parent.postMessage(message, e.origin)
+            break
+          case 'remove':
+            document.cookie = document.cookie
```

Comment:
I see that you already implemented the getCookies() function and the map object is now being used in most places, but is there a reason to parse and iterate the cookies again in this line?

Reply:
Easier to not have to rebuild all the cookie syntax from getCookies. Less complexity and I didn't mind inefficiency of something that leads to logging out.